developer/src/kmc/src/messages/messages.ts

```diff
@@ -0,0 +1,39 @@
+import { CompilerErrorNamespace, CompilerErrorSeverity, CompilerMessageSpec as m } from "@keymanapp/common-types";
+
+const Namespace = CompilerErrorNamespace.Infrastructure;
+const SevInfo = CompilerErrorSeverity.Info | Namespace;
+// const SevHint = CompilerErrorSeverity.Hint | Namespace;
+// const SevWarn = CompilerErrorSeverity.Warn | Namespace;
+const SevError = CompilerErrorSeverity.Error | Namespace;
+const SevFatal = CompilerErrorSeverity.Fatal | Namespace;
+
+export class InfrastructureMessages {
+  static Fatal_UnexpectedException = (o:{e: any}) => m(this.FATAL_UnexpectedException,
+    `Unexpected exception: ${(o.e ?? 'unknown error').toString()}\n\nCall stack:\n${(o.e instanceof Error ? o.e.stack : (new Error()).stack)}`);
+  static FATAL_UnexpectedException = SevFatal | 0x0001;
+
+  static Info_BuildingFile = (o:{filename:string}) => m(this.INFO_BuildingFile,
+    `Building ${o.filename}`);
+  static INFO_BuildingFile = SevInfo | 0x0002;
+
+  static Error_FileDoesNotExist = (o:{filename:string}) => m(this.ERROR_FileDoesNotExist,
+    `File ${o.filename} does not exist`);
+  static ERROR_FileDoesNotExist = SevError | 0x0003;
+
+  static Error_FileTypeNotRecognized = (o:{filename: string, extensions: string}) => m(this.ERROR_FileTypeNotRecognized,
+    `Unrecognised input file ${o.filename}, expecting ${o.extensions}, or project folder`);
+  static ERROR_FileTypeNotRecognized = SevError | 0x0004;
+
+  static Error_OutFileNotValidForProjects = () => m(this.ERROR_OutFileNotValidForProjects,
+    `--out-file should not be specified for project builds`);
+  static ERROR_OutFileNotValidForProjects = SevError | 0x0005;
+
+  static Info_FileBuiltSuccessfully = (o:{filename:string}) => m(this.INFO_FileBuiltSuccessfully,
+    `${o.filename} built successfully.`);
+  static INFO_FileBuiltSuccessfully = SevInfo | 0x0006;
```

Comment:
With the status masking, wouldn't `SevInfo | 0x0003` be available?

Is this the expected status for successful compile/build?

Reply:
* All base codes are unique (when | with their namespace) -- it's just safer to always do this than to accidentally end up using the same code twice (also, in theory we might want to increase/decrease severity of a code in future?)
* This is an informational message. The return status for kmc will be 0 on success, 1 on failure (there may be other statuses in future).